js/build.js

```diff
@@ -22,18 +22,106 @@ Fliplet.Widget.instance('login-ds', function(data) {
   var DATA_DIRECTORY_EMAIL_COLUMN = data.emailColumn;
   var DATA_DIRECTORY_PASS_COLUMN = data.passColumn;
   var ORG_NAME = Fliplet.Env.get('organizationName');
+  var defaultLang = {
+    en: {
+      'component.login': {
+        title: 'Login',
+        instruction: 'Enter your access details below',
+        emailPlaceholder: 'Enter your email',
+        passwordPlaceholder: 'Enter your password',
+        mismatch: 'The details you entered don’t match our records. Please try again.',
+        action: 'Log in',
+        loader: 'Loading...',
+        forgotPassword: 'Forgot your password?',
+      },
+      'component.login.restore': {
+        instruction: 'To reset your password you must verify your email address.',
+        mismatch: 'Your email address cannot be found. Please try again.',
+        goBack: 'Back to Login',
+        action: 'Verify email',
+        verifyCode: 'I have a verification code',
+      },
+      'component.login.reset': {
+        instruction: 'Enter your new password below.',
+        passwordPlaceholder: 'Enter your new password',
+        confirmPasswordPlaceholder: 'Confirm your new password',
+        mismatch: 'Passwords don\'t match. Please try again.',
+        action: 'Reset password',
+      },
+      'component.login.verify': {
+        instruction: 'A verification code was sent to',
+        codePlaceholder: 'Enter verification code',
+        mismatch: 'You entered the wrong code or the code expired. Please try again or request a new code.',
+        resend: 'Resend verification code',
+        action: 'Verify',
+        sendCode: 'A new code was sent to your email.',
+      },
+      'component.login.allDone': {
+        instruction: 'You can now login using your email address and new password.',
+        action: 'Login'
+      }
+    }
+  };
+  var currentLang = navigator.language;
 
   if (Fliplet.Navigate.query.error) {
     $container.find('.login-error').html(Fliplet.Navigate.query.error).removeClass('hidden');
   }
 
+  function initI18Next() {
+    i18next.init({
+      lng: currentLang,
+      fallbackLng: 'en',
+      resources: defaultLang 
+    }).then(function() {
+      setCurrentLang()
+
+      $container.css('visibility', 'visible');
+      // Fire hook for other features to adjust UI render if necessary
+      Fliplet.Hooks.run('i10nComplete');
```

Comment:
I'd stay await from the i18n word, clients should only know that this runs when the localisation has been completed (e.g. `localized` or `localizeCompleted`). Furthermore, we need to pass the component as second argument so the user can check which component has finished loading.

Reply:
@squallstar `loginLocalized` will it be okay hook name?

Reply:
Yes, that works although I am wondering who is going to use this hook? Is it part of the requirements? And if so, why not adding a generic hook when the component is ready?

Reply:
@squallstar 

>  I am wondering who is going to use this hook?

As I understand it will be used by developers threw the `Developers options` in the studio.

> Is it part of the requirements?

Yes, it is a part of the requirement that is lined in the [task](https://github.com/Fliplet/fliplet-studio/issues/6595).

> And if so, why not adding a generic hook when the component is ready?

Well, in the ideal situation we should get translation from the users before we render component but we couldn't do it with handlebars as it was lined [here](https://github.com/Fliplet/fliplet-studio/issues/6595#issuecomment-648099269) and [here ](https://github.com/Fliplet/fliplet-studio/issues/6595#issuecomment-648264598).
Yes, our current implementation is allowing as fire this hook after the component was rendered. But to avoid glitch which may occur we setting `visibility: hidden` to the component as it was suggested [here](https://github.com/Fliplet/fliplet-studio/issues/6595#issuecomment-648790455) and set `visibility: visible`  when text is set according to the browser language. That is why firing this hook as component ready isn't true because at the moment when this hook is fired component will be hidden by `visibility: hidden`.